sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,530 @@
+|===
+
+By calling `async` on a `task_sequence` object more than once, the user implies
+that those invocations of `f` can be run in parallel. The implementation is,
+however, not obligated to run the invocations in parallel except in so far as is
+necessary to meet the forward progress guarantees outlined in the section on
+Progress Guarantees.
+
+The `async` function call is non-blocking in that it may return before the
+asynchronous `f` invocation completes executing, and potentially before `f` even
+begins executing (return from the `async` provides no implicit information on
+the execution status of `f`).
+
+An asynchronous invocation of `f`, upon the end of its execution of `f`, will
+produce a result. At that point, that particular asynchronous invocation is now
+considered to be complete, and the result of that invocation is now ready to be
+retrieved by a `get` call. The result of each completed invocation is stored in
+a results data structure until the appropriate `get` call retrieves it. This
+results data structure will keep the results in the same order in which their
+corresponding `async` invocations were invoked. The `get` call retrieves results
+from this results data structure. The `get` call blocks if there is no result to
+retrieve.
+
+Both `async` and `get` functions may only be invoked on the device on which a
+`task_sequence` object has been instantiated. Calling `async` or `get` on a
+different device results in undefined behavior.
+
+=== `task_sequence` Scoping
+
+`task_sequence` objects should retire all outstanding `async` invocations before
+exiting scope. This is performed by the `task_sequence` destructor, by calling
+`get` on all outstanding invocations and blocking destruction of the object
+until all invocations are completed. The exception to this case is if the
+`balanced` property was specified, in which case it is the programmer's
+responsibility to ensure that corresponding `get` has been called for all
+invocations launched using `async` before the lifetime of the `task_sequence`
+object ends.
+
+It is expected that many common coding patterns will guarantee that the number
+of `async` and `get` calls match (are balanced) before a `task_sequence` object
+is destroyed. To provide more information to the compiler and to relax the
+requirement for `get` to be invoked implicitly, the property `balanced` may be
+specified on a `task_sequence` object, which guarantees that a user will not
+allow a destructor on that `task_sequence` object to be called when there are
+outstanding `async` invocations that have not been balanced by a matching `get`
+call. On spatial architectures, in the presence of this property, potentially
+expensive hardware may be elided. It is undefined behavior to specify the
+`balanced` property on `task_sequence` and then to allow the `task_sequence`
+object to be destroyed while there are any `async` invocations for which `get`
+has not been called.
```

Comment:
The end of the first paragraph and the beginning of the second are redundant.

Reply:
removed